src/main/java/com/gooddata/dataset/DatasetException.java

```diff
@@ -5,23 +5,36 @@
 
 import com.gooddata.GoodDataException;
 
+import java.util.Collection;
+
+import static java.util.Arrays.asList;
+
 /**
  * Represents error in DatasetService
  */
 public class DatasetException extends GoodDataException {
 
-    private final String dataset;
+    private final Collection<String> datasets;
 
     public DatasetException(String message, String dataset) {
         this(message, dataset, null);
     }
 
     public DatasetException(String message, String dataset, Throwable cause) {
-        super("Load dataset " + dataset + " failed: " + message, cause);
-        this.dataset = dataset;
+        super("Load dataset " + asList(dataset) + " failed: " + message, cause);
+        this.datasets = asList(dataset);
+    }
+
+    public DatasetException(String message, Collection<String> datasets, Throwable cause) {
+        super("Load datasets " + datasets + " failed: " + message, cause);
+        this.datasets = datasets;
+    }
+
+    public DatasetException(String message, Collection<String> datasets) {
+        this(message, datasets, null);
     }
 
-    public String getDataset() {
-        return dataset;
+    public Collection<String> getDatasets() {
```

Comment:
preserve `public String getDataset()` to retain backward compatibility. eg. return `datasets.toString()`